tests/python/pants_test/reporting/test_reporting_integration.py

```diff
@@ -265,6 +268,70 @@ def test_zipkin_reporter_for_v2_engine(self):
         v2_span_name_part, trace
         ))
 
+  def run_mock_server(self, bin_name, arguments):
```

Comment:
If you want to get really fancy, you could make this into a `@contextmanager`. Something like:

```
@contextmanager
def run_server(...):
  try:
    <all the code that you have now>
    yield (process, port)
  finally:
    process.communicate('\n')
```

Then, the lifetime of servers is done for you, and the test could be shortened as:
```
with run_cas_server() as (cas_server, cas_server_port), \
        run_execution_server(...) as (execution_server, execution_server_port):
    # The actual test code...
    ZipkinHandler = zipkin_handler()
    random_store_location = secrets.token_hex(8)
    ...
```

More info on context managers: https://alysivji.github.io/managing-resources-with-context-managers-pythonic.html